tower-grpc-build/src/lib.rs

```diff
@@ -176,3 +179,25 @@ fn unqualified(ty: &str, level: usize) -> String {
 
     v.join("::")
 }
+
+
+/// Converts a `snake_case` identifier to an `UpperCamel` case Rust type
+/// identifier.
+///
+/// This is identical to the same [function] in `prost-build`, however, we
+/// reproduce it here as `prost` does not publically export it.
+///
+/// We need this as `prost-build` will only give us the snake-case transformed
+/// names for gRPC methods, but we need to use method names in types as well.
+///
+/// [function]: https://github.com/danburkert/prost/blob/d3b971ccd90df35d16069753d52289c0c85014e4/prost-build/src/ident.rs#L28-L38
+fn to_upper_camel(s: &str) -> String {
```

Comment:
Having to implement this ourselves is unfortunate as it makes us brittle to changes in prost.

Maybe prost could include additional variants in the structs it provides?

Thoughts @danburkert?

Reply:
I don't think this is particularly brittle WRT changes in prost, since prost never exports a `to_upper_camel`ized version of a method name.  This appears to be completely specific to grpc, so I'm not in favor of exporting it from prost-build.  Perhaps I'm missing something, though.